Gems/ScriptCanvas/Code/Include/ScriptCanvas/Libraries/Core/AzEventHandler.cpp

```diff
@@ -14,6 +18,148 @@
 #include <ScriptCanvas/Utils/BehaviorContextUtils.h>
 #include <ScriptCanvas/Utils/NodeUtils.h>
 
+namespace AzEventHandlerCpp
+{
+    using namespace ScriptCanvas;
+    using namespace ScriptCanvas::Nodes;
+    using namespace ScriptCanvas::Nodes::Core;
+    
+    struct AzEventEntryData_v0
+    {
+        static void Reflect(AZ::ReflectContext* context)
+        {
+            if (auto serializeContext = azrtti_cast<AZ::SerializeContext*>(context); serializeContext != nullptr)
+            {
+                serializeContext->Class<AzEventEntryData_v0>()
+                    ->Field("eventName", &AzEventEntryData_v0::m_eventName)
+                    ->Field("parameterSlotIds", &AzEventEntryData_v0::m_parameterSlotIds)
+                    ->Field("parameterNames", &AzEventEntryData_v0::m_parameterNames)
+                    ->Field("azEventInputSlotId", &AzEventEntryData_v0::m_azEventInputSlotId)
+                    ;
+            }
+        }
```

Comment:
Isn't there still a problem with the [InitEventFromMethod](https://github.com/o3de/o3de/blob/48492e9b5c67f5ef3a3f98820ff7374c81218dc0/Gems/ScriptCanvas/Code/Include/ScriptCanvas/Libraries/Core/AzEventHandler.cpp#L217) function below appending to the `m_parameterSlotIds` vector?

Reply:
Doing a search that method is only invoked once when the AzEventHandler node is created from the [Node Palette](https://github.com/o3de/o3de/blob/development/Gems/ScriptCanvas/Code/Editor/Nodes/NodeCreateUtils.cpp#L437) and once when the Translation Generation class creates an AzEventHandler node.

So the code is fine as long the InitFromEventMethod isn't being invoked after serializing in the node.

Reply:
Yeah, it's not an issue, I verified that as well.